src/main/java/io/burt/athena/result/s3/InputStreamResponseTransformer.java

```diff
@@ -28,12 +30,15 @@
     private Throwable error;
     private AtomicBoolean complete;
     private AtomicInteger approximateBufferSize;
+    private AtomicInteger requests;
+    private volatile float approximateChunkSize;
```

Comment:
This is not initialised anywhere which is dangerous. Setting it to the length of the first chunk received should be a good starting point.

Reply:
It would be nontrivial to initialize this from the first chunk (and also ensure that it is nonzero), so I think the better option is to initialize it to something. I picked 8192 completely arbitrarily, but with the continued adjustment, it shouldn't matter so much, but it should not be too small.